[BUG] Singular name is derived, although a @singular annotation is present

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Nature of Your Project

TypeScript

### Current Behavior

When I try to enter the singular and plural names as described in the documentation, I get the error message that the derived singular already exists.
`[ERROR] Derived singular 'SalesDelivery' for your entity 'my.bookshop.SalesDelivery', already exists. The resulting types will be erronous. Consider using '@singular:'/ '@plural:' annotations in your model or move the offending declarations into different namespaces to resolve this collision.`

### Expected Behavior

The name of the singular entity should be taken from the @singular annotation and not derived.

### Steps To Reproduce

```cds
@singular: 'SalesDelivery'
@plural  : 'SalesDeliveries'
entity SalesDelivery {
    deliveryNumber : String(10);
}
```

### Environment

```markdown
| @cap-js/cds-typer      | 0.21.0                                                                  |
| @cap-js/cds-types      | 0.2.0                                                                   |
| @cap-js/sqlite         | 1.7.1                                                                   |
| @sap/cds               | 7.9.2                                                                   |
| @sap/cds-compiler      | 4.9.4                                                                   |
| @sap/cds-dk (global)   | 7.9.2                                                                   |
| @sap/cds-fiori         | 1.2.3                                                                   |
| @sap/cds-foss          | 5.0.0                                                                   |
| @sap/cds-mtxs          | 1.18.0                                                                  |
| @sap/eslint-plugin-cds | 3.0.3                                                                   |
| Node.js                | v18.18.0
```


### Repository Containing a Minimal Reproducible Example

_No response_

### Anything else?

_No response_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,9 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 The format is based on [Keep a Changelog](http://keepachangelog.com/).
 
 ## Version 0.24.0 - TBD
+### Fixed
+- Suppressed an error that would incorrectly point out naming clashes when an entity was named in singular inflection in the model
+
 
 ## Version 0.23.0 - 2024-07-04
 ### Fixed

--- a/lib/visitor.js
+++ b/lib/visitor.js
@@ -250,7 +250,9 @@ class Visitor {
         }
 
         // as types are not inflected, their singular will always clash and there is also no plural for them anyway -> skip
-        if (!isType(entity) && `${ns.asNamespace()}.${singular}` in this.csn.xtended.definitions) {
+        // if the user defined their entities in singular form we would also have a false positive here -> skip
+        const namespacedSingular = `${ns.asNamespace()}.${singular}`
+        if (!isType(entity) && namespacedSingular !== fq && namespacedSingular in this.csn.xtended.definitions) {
             LOG.error(
                 `Derived singular '${singular}' for your entity '${fq}', already exists. The resulting types will be erronous. Consider using '@singular:'/ '@plural:' annotations in your model or move the offending declarations into different namespaces to resolve this collision.`
             )

